realm/realm-library/src/androidTest/kotlin/io/realm/ManagedSetTester.kt

```diff
@@ -643,12 +643,121 @@ class ManagedSetTester<T : Any>(
         }
     }
 
+    override fun addRealmChangeListener() {
+        looperThread.runBlocking {
+            val looperThreadRealm = Realm.getInstance(config)
+
+            // Get set
+            val set = initAndAssertEmptySet(looperThreadRealm)
+
+            // Define operation we perform on the dictionary
+            var operation = ChangeListenerOperation.UNDEFINED
+
+            set.addChangeListener { newSet ->
+                when (operation) {
+                    ChangeListenerOperation.INSERT -> {
+                        assertEquals(initializedSet.size, newSet.size)
```

Comment:
If you post a runnable to the looper thread it will not be executed until the current "block" execution is done, and that interferes with the test logic in the sense that I'm already blocking the execution of the JUnit test runner by using `looperThread.runBlocking {}`. This means the runnable would be executed after the block is done and thus the test wouldn't work.

The only way I see I could do as you suggest is by testing individual operations separately and adding different change listeners inside different `looperThread.runBlocking {}` blocks, which I think is a bit meh to do...